engine/src/main/java/org/pentaho/di/trans/steps/textfileoutput/TextFileOutput.java

```diff
@@ -639,6 +639,9 @@ private void writeField( ValueMetaInterface v, Object valueData, byte[] nullStri
               && containsSeparatorOrEnclosure( str, data.binarySeparator, data.binaryEnclosure ) ) {
             writeEnclosures = true;
           }
+        }  else if (v.isNumeric() && !meta.isEnclosureFixDisabled()
```

Comment:
As mentioned in the jira description.
https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/main/java/org/pentaho/di/trans/steps/textfileoutput/TextFileOutput.java#L635-L647
and 
https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/main/java/org/pentaho/di/trans/steps/textfileoutput/TextFileOutput.java#L732-L738

share the exact same logic. There needs to be a refactor, D-R-Y (Don't Repeat Yourself).